lib/rust/types/src/unit.rs

```diff
@@ -1,874 +1,940 @@
-//! # DEPRECATED!!!
-//! THIS MODULE IS DEPRECATED. USE `unit2` INSTEAD.
-//!
-//! Defines utilities for creating custom strongly typed units. For example, unit `Angle` could be a
-//! wrapper for `f32`.
+//! Utilities for creating custom strongly typed units. For example, unit `Duration` is a wrapper
+//! for [`f32`] and defines time-related utilities.
 //!
 //! Units automatically implement a lot of traits in a generic fashion, so you can for example add
-//! to angles together, or divide angle by a number, but you are not allowed to divide a number by
-//! an angle. Rarely you may want to use very custom rules for unit definition. In such a case, you
-//! should use other macros defined in this module. Look at the definitions below to learn the
-//! usage patterns.
-//!
-//! ## Why Macros
-//! You may wonder why this utility is defined as a macro that generates hundreds of lines of code
-//! instead of a generic type `Unit<Phantom,Repr>`. The later approach has one major issue. The
-//! definition `type Angle = Unit<AngleType,f32>` will be dysfunctional, as you would not be allowed
-//! to implement many impls because the type `Unit` would be defined in an external crate.
-//!
-//! ## Macro Naming Pattern
-//! Many macros defined in this module use a specific naming, for example
-//! `impl_UNIT_x_FIELD_to_UNIT`. You should interpret it as a macro defining a transformation
-//! similar to `fn<Unit,Field>(unit:Unit, field:Field) -> Unit`, where `Field` is a specific field
-//! of the `Unit` (in most cases units have single field).
+//! [`Duration`] together, or divide one [`Duration`] by a number or another [`Duration`] and get
+//! [`Duration`] or a number, respectfully. You are allowed to define any combination of operators
+//! and rules of how the result inference should be performed.
 
```

Comment:
So, I guess we don't keep the deprecated module, but replace it with unit2, and just the GH does not understand it?